src/hardware/imf.cpp

```diff
@@ -0,0 +1,13422 @@
+	a   Note ON/OFF                         [80]~[9F]                           [00]~[2F]
+	b   After Touch and Pitchbend           [A0]~[AF], [D0]~[DF], [E0]~[EF]     [50]~[5F], [60]~[6F]
+	c   Control Change                      [B0]~[BF]                           [30]~[3F]
+		Program Change                      [C0]~[CD]                           [40]~[4F]
+	d   System Exclusive                    [F0], [F7]
+		System Common                       [F1], [F2], [F3], [F6]              [70]~[7F]
+	e   System Real-Time                    [F8], [FA]~[FC]
+	f   ??? only to SP                      N/A                                 N/A
+
+	Note: Undefined MIDI status bytes ([F4], [F5], [F9], [FD], [FF]) and the active sensing code ([FE]) not listed
+	      in the preceding table are blocked. Succeeding data bytes are also blocked.
+*/
+// clang-format on
+
+#pragma pack(push, 1)
+struct MidiFlowPath {
+	// This is the path from MIDI IN to the system. This path is enabled
+	// when the data received at MIDI IN is to be processed by the system.
+	uint8_t MidiIn_To_System = 0;
+	// This is the patch from the system to MIDI OUT. This path is used when
+	// sending a message from the system to an external MIDI device. If this
+	// path is enabled, it is automatically closed when CHAIN mode is set.
+	// When CHAIN mode is cancelled, the path returns to its previous state.
+	// Even if this path has been closed by entry into CHAIN mode, status
+	// reports from the music card will reflect its programmed state.
+	uint8_t System_To_MidiOut = 0;
+	// This is the path from MIDI IN to the sound processor. This path is set
+	// so that the music card can be controlled by an external MIDI devince.
+	uint8_t MidiIn_To_SP = 0;
+	// This is the route from the system to the sound processor. This path
+	// is set so that the music card can be controlled by the processor. Note:
+	// If you try to set both the "MIDI IN -> Sound Processor" and "System
+	// -> Sound Processor" paths to pass system real-time messages,
+	//       only the "MIDI IN -> Sound Processor" path is set. The other
+	//       path is ignored with no error report message.
+	uint8_t System_To_SP = 0;
+	// This is the path from MIDI IN to MIDI OUT. When this path is set, the
+	// MIDI OUT terminal functions in essentially the same manner as the MIDI
+	// THRU terminal. When the "System -> MIDI OUT" path is concurrently set,
+	// the data from the system and the data input to MIDI IN can be merged
+	// in MIDI OUT. When this path is set, the program request from a MIDI
+	// line is not valid, and a MIDI handshaking message cannot be output to
+	// a MIDI line. This path automatically closes when CHAIN mode is set.
+	// When CHAIN mode is cancelled, this path returns  to its previous state.
+	// Even if this path has been closed by entry into CHAIN mode, status
+	// reports from the music card reflect its programmed state.
+	uint8_t MidiIn_To_MidiOut = 0;
```

Comment:
Add new lines for readability, remove underscores from names.

Reply:
Fixed up whitespace, good catch @johnnovak - those variables were swallowed up around their descriptions.